File: src/TaskManagement.Application.Contracts/Assignments/CreateAssignmentInput.cs
using TaskManagement.Domain.Assignments;

namespace TaskManagement.Application.Contracts.Assignments
{
    public class CreateAssignmentInput
    {
        public Guid AssignedUserId { get; set; }

        public DateTime Deadline { get; set; }

        public string Name { get; set; } = string.Empty;

        public AssignmentPriority Priority { get; set; }

        public AssignmentSection Section { get; set; }

        public AssignmentAlertType? AlertType { get; set; }

        public string? Description { get; set; }
    }
}

File: src/TaskManagement.Web/src/services/assignments/types/create-assignment-input.ts
import { AssignmentAlertType } from './assignment-alert-type'
import { AssignmentPriority } from './assignment-priority'
import { AssignmentSection } from './assignment-section'

export interface CreateAssignmentInput {
    assignedUserId: string
    deadline: string
    name: string
    priority: AssignmentPriority
    section: AssignmentSection
    alertType?: AssignmentAlertType
    description?: string
}
